worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -907,15 +907,18 @@ private[celeborn] class Worker(
 
     def waitTime: Long = waitTimes * interval
 
-    while (!storageManager.shuffleKeySet().isEmpty && waitTime < timeout) {
+    val unreleasedShuffleKeys = storageManager.shuffleKeySet()
+    while (!unreleasedShuffleKeys.isEmpty && waitTime < timeout) {
       Thread.sleep(interval)
       waitTimes += 1
     }
-    if (storageManager.shuffleKeySet().isEmpty) {
+
+    if (unreleasedShuffleKeys.isEmpty) {
       logInfo(s"Waiting for all shuffle expired cost ${waitTime}ms.")
     } else {
       logWarning(s"Waiting for all shuffle expired cost ${waitTime}ms, " +
-        s"unreleased shuffle: \n${storageManager.shuffleKeySet().asScala.mkString("[", ", ", "]")}")
+        s"unreleased shuffle: \n${unreleasedShuffleKeys.asScala.mkString("[", ", ", "]")}")
+      workerSource.incCounter(UNRELEASED_SHUFFLE_COUNT, unreleasedShuffleKeys.size)
```

Comment:
In any scenarios would we dec count this value?

Reply:
no, the shuffle data is lost.

Reply:
@s0nskar, why not use gauge to metric unreleased shuffle count?

Reply:
@SteNicholas As this is a ever increasing metric, IMO counter is more appropriate for this.

https://prometheus.io/docs/concepts/metric_types/

Reply:
If a worker is decommissioned, the worker will be shut down. Using a gauge here will be reasonable. The metric won't be increasing because the worker is gone.

Reply:
> If a worker is decommissioned, the worker will be shut down. Using a gauge here will be reasonable. The metric won't be increasing because the worker is gone.
@s0nskar, WDYT?

Reply:
> If a worker is decommissioned, the worker will be shut down. Using a gauge here will be reasonable. The metric won't be increasing because the worker is gone.

@s0nskar, wdyt?

Reply:
Sorry for the late reply, just had a very busy oncall week. Converted the metric to gauge.